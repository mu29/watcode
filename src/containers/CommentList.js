import React from 'react'
import { connect } from 'react-redux'
import { CommentList } from 'components'

const CommentListContainer = props => <CommentList { ...props } />

const mapStateToProps = () => ({
  list: [{
    id: 1,
    content: '달인급 캐릭터 분량 조절의 기술. 이 긴 영화에 지루함 한 톨 없다. 너는 어른이 다 됐더라, 피터',
    userAttributes: {
      name: '영화번역가 황석희',
    },
  }, {
    id: 2,
    content: '마블의 모든 것을 쏟아부은 명작',
    userAttributes: {
      name: '이동진',
    },
  }, {
    id: 3,
    content: '선발대 확인완료 돌격하라!!',
    userAttributes: {
      name: '뮤',
    },
  }],
})

export default connect(mapStateToProps)(CommentListContainer)
