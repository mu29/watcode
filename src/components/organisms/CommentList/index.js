import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { CommentItem } from 'components'
import { CommentForm } from 'containers'
import { palette } from 'services/style'

const Wrapper = styled.div`
  padding: 1rem;
  background-color: ${palette('gray.20')};
`

const StyledCommentItem = styled(CommentItem)`
  margin-top: 1rem;
`

const CommentList = ({ list }) => (
  <Wrapper>
    <CommentForm />
    { list.map(item => <StyledCommentItem key={ item.id } comment={ item } />) }
  </Wrapper>
)

CommentList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default CommentList
