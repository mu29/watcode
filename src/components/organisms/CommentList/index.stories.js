import React from 'react'
import { storiesOf } from '@storybook/react'
import CommentList from '.'

const comments = [{
  id: 1,
  content: '달인급 캐릭터 분량 조절의 기술. 이 긴 영화에 지루함 한 톨 없다. 너는 어른이 다 됐더라, 피터',
  userAttributes: {
    name: '영화번역가 황석희',
  },
}, {
  id: 2,
  content: '마블의 모든 것을 쏟아부은 명작',
  userAttributes: {
    name: '이동진',
  },
}, {
  id: 3,
  content: '선발대 확인완료 돌격하라!!',
  userAttributes: {
    name: '뮤',
  },
}]

storiesOf('Organisms/CommentList', module)
  .add('기본', () => (
    <CommentList list={ comments } />
  ))
