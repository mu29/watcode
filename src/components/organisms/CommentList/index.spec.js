import React from 'react'
import CommentList from '.'

const comments = [{
  id: 1,
  content: '달인급 캐릭터 분량 조절의 기술. 이 긴 영화에 지루함 한 톨 없다. 너는 어른이 다 됐더라, 피터',
  userAttributes: {
    name: '영화번역가 황석희',
  },
}, {
  id: 2,
  content: '마블의 모든 것을 쏟아부은 명작',
  userAttributes: {
    name: '이동진',
  },
}, {
  id: 3,
  content: '선발대 확인완료 돌격하라!!',
  userAttributes: {
    name: '뮤',
  },
}]

const component = (props = {}) => withTheme(<CommentList list={ comments } { ...props } />)
const wrap = (props = {}) => shallow(component(props)).dive()

describe('<CommentList />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component())
    expect(wrapper).toMatchSnapshot()
  })

  it('3개의 아이템을 그려준다', () => {
    const wrapper = wrap()
    expect(wrapper.find('StyledCommentItem')).toHaveLength(3)
    const firstItem = wrapper.childAt(0).dive()
    expect(firstItem).toMatchSnapshot()
  })
})
