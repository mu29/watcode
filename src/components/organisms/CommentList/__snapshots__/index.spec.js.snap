// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentList /> 3개의 아이템을 그려준다 1`] = `
<span
  componentname="CommentForm"
/>
`;

exports[`<CommentList /> 정상적으로 렌더링된다 1`] = `
.c0 {
  padding: 1rem;
}

<div
  class="c0"
>
  <span
    componentname="CommentForm"
  />
  <span
    componentname="CommentItem"
  />
  <span
    componentname="CommentItem"
  />
  <span
    componentname="CommentItem"
  />
</div>
`;
